### ISSUE TITLE
test dataset publishing

### ISSUE THREAD

**Issue by amueller** [openml/openml-python#113]
downloading and publishing iris again doesn't work (maybe string issues in the description?)

**Comment by PGijsbers** [comment]
Still relevant, MVE:
```
import openml
iris = openml.dataset.get_dataset(61)
iris.publish()
```

output:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "D:\repositories\openml-python\openml\datasets\dataset.py", line 436, in publish
    file_elements=file_elements)
  File "D:\repositories\openml-python\openml\_api_calls.py", line 52, in _perform_api_call
    file_elements=file_elements)
  File "D:\repositories\openml-python\openml\_api_calls.py", line 98, in _read_url_files
    raise _parse_server_exception(response, url=url)
openml.exceptions.OpenMLServerException: Problem validating uploaded description file
```

**Comment by ArlindKadra** [comment]
@mfeurer @amueller I added a fix in  #440. @PGijsbers  the code above would still throw an error, as it is trying to upload an url to the dataset and also the dataset object. One of them should be removed.

```
import openml
iris = openml.datasets.get_dataset(61)
iris.url = None
iris.publish()
```

**Comment by ArlindKadra** [comment]
Solved in #440
